src/displayBackend/wayland/Surface.cpp

```diff
@@ -24,25 +28,33 @@ namespace Wayland {
  * Surface
  ******************************************************************************/
 
-Surface::Surface(wl_compositor* compositor) :
```

Comment:
Why not pass `wl_compositor` and `wp_presentation` as parameters? IMHO, passing a raw pointer to `Compositor` is not safe.

Reply:
wp_presentation is not ready at this moment.  "passing a raw pointer to Compositor" should I use some smart  ptrs here?